test/verify/check-system-s4u-ssh

```diff
@@ -90,8 +94,33 @@ gssapi.creds.rcred_cred_store.store_cred_into({ 'ccache'.encode(): 'FILE:/etc/co
         # enable gssapiauth
         sshd_machine.execute("sed -ri 's/#GSSAPIAuthentication (yes|no)/GSSAPIAuthentication yes/' /etc/ssh/sshd_config")
         sshd_machine.execute("systemctl daemon-reload && systemctl restart sshd")
+        client_machine.execute("KRB5RCACHEDIR=\"/var/cache/krb5rcache\" KRB5_TRACE=\"/dev/stderr\" KRB5CCNAME=\"/etc/cockpit.ccache\" ssh -vvv -K  user@sshserver.cockpit.lan echo hello")
 
-        client_machine.execute("KRB5_TRACE=\"/dev/stderr\" KRB5CCNAME=\"/etc/cockpit.ccache\" ssh -vvv -K  user@sshserver.cockpit.lan echo hello")
+        # configure gssapi authentication for sudo
+        sshd_machine.write('/etc/sudoers.d/cockpit-krb5-sudo', """
+Defaults    env_keep += "KRB5CCNAME"
+        """)
+
+        sshd_machine.write('/etc/pam.d/sudo', """
+auth       sufficient   pam_sss_gss.so
+auth       required     pam_deny.so
+account    include      system-auth
+password   include      system-auth
+session    optional     pam_keyinit.so revoke
+session    required     pam_limits.so
+session    include      system-auth
+""")
+
+        sshd_machine.execute(script="""#!/bin/sh
+        sed -i '/\[domain\/cockpit.lan\]/ a pam_gssapi_services = sudo, sudo-i' /etc/sssd/sssd.conf
+        systemctl restart sssd
+        usermod -G wheel user
+        su user
+        echo -ne 'foobarfoo\nfoobarfoo\nfoobarfoo\n' | kinit -c /home/user/cache user
+        KRB5CCNAME=/home/user/cache sudo echo hai
+        exit 0
+        """)
+        client_machine.execute("KRB5_TRACE=\"/dev/stderr\" KRB5CCNAME=\"/etc/cockpit.ccache\" ssh -vvv -K  user@sshserver.cockpit.lan bash -c 'KRB5CCNAME=/home/user/cache sudo echo hai'")
```

Comment:
Could you please run `sudo whoami` instead, and check that the returned output contains `root`? That would ensure that it actually worked.

I don't quite understand the difference between the script execution above, and this line? They seem to do similar things?

Reply:
I'll do that, the second ssh-ing from the client machine via gssapi auth and then using the cache built above to sudo, putting the 2 things together. But it's indeed just testing the same thing.